modules/tinymce/src/themes/silver/test/ts/browser/window/SilverDialogSliderApiTest.ts

```diff
@@ -0,0 +1,58 @@
+import { FocusTools, Keys, TestStore } from '@ephox/agar';
+import { context, describe, it } from '@ephox/bedrock-client';
+import { Arr } from '@ephox/katamari';
+import { SugarDocument } from '@ephox/sugar';
+import { TinyHooks, TinyUiActions } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import { Dialog } from 'tinymce/core/api/ui/Ui';
+
+import * as DialogUtils from '../../module/DialogUtils';
+
+describe('browser.tinymce.themes.silver.window.SilverDialogSliderApiTest', () => {
+  const store = TestStore();
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    base_url: '/project/tinymce/js/tinymce'
+  }, []);
+
+  const dialogSpec: Dialog.DialogSpec<{ sliderField: number }> = {
+    title: 'Silver Dialog Test Slider Api',
+    body: {
+      type: 'panel',
+      items: [
+        {
+          type: 'slider',
+          name: 'sliderField',
+          label: 'Slider label',
+          min: 0,
+          max: 100
+        }
+      ]
+    },
+    initialData: {
+      sliderField: 50
+    },
+    onChange: () => {
+      store.add('logOnChangeEvent');
+    }
+  };
+
+  Arr.each([
+    { label: 'Modal', params: { }},
+    { label: 'Inline', params: { inline: 'toolbar' as 'toolbar' }}
+  ], (test) => {
+    context(test.label, () => {
+      it('TINY-10428: dialog onChange handler should fire when using arrow key on slider', async () => {
+        store.clear();
+        const editor = hook.editor();
+        DialogUtils.open(editor, dialogSpec, test.params);
+        await TinyUiActions.pWaitForDialog(editor);
+
+        await FocusTools.pTryOnSelector('Focus should be on slider rail', SugarDocument.getDocument(), '.tox-slider__rail');
+        TinyUiActions.keydown(editor, Keys.left());
+
+        store.assertEq('Check if on change event is logged after using arrow key', [ 'logOnChangeEvent' ]);
```

Comment:
Please check the value changed, rather than just that an event fired. Also check both left and right key directions for the correct value change.

- [ ] improve tests

Reply:
There was no issue with value change. Slider just didn't emit `formChangeEvent` on that change. Testing if the value changed unfortunately doesn't cover my changes. Testing if the event was logged in the dialog is what actually tests my changes. 
All of that aside, I think it is good to assert both value and event to make sure arrow keys work as expected. Adding assertion to test both value change and event fired for left and right key 👍

Reply:
yes, while it's not technically part of your change it is the reason the event fires, so I think it's worth checking here.